packages/react-core/src/components/Tooltip/examples/TooltipOptions.tsx

```diff
@@ -0,0 +1,205 @@
+          <Select
+            onToggle={() => setPositionSelectOpen(!positionSelectOpen)}
+            onSelect={(_event, selection) => {
+              setPosition(selection);
+              setPositionSelectOpen(false);
+            }}
+            isOpen={positionSelectOpen}
+            selections={position}
+            menuAppendTo={() => document.body}
+          >
+            <SelectOption value={TooltipPosition.auto} />
+            <SelectOption value={TooltipPosition.top} />
+            <SelectOption value={TooltipPosition.bottom} />
+            <SelectOption value={TooltipPosition.left} />
+            <SelectOption value={TooltipPosition.right} />
+            <SelectOption value={TooltipPosition.topStart} />
+            <SelectOption value={TooltipPosition.topEnd} />
+            <SelectOption value={TooltipPosition.bottomStart} />
+            <SelectOption value={TooltipPosition.bottomEnd} />
+            <SelectOption value={TooltipPosition.leftStart} />
+            <SelectOption value={TooltipPosition.leftEnd} />
+            <SelectOption value={TooltipPosition.rightStart} />
+            <SelectOption value={TooltipPosition.rightEnd} />
+          </Select>
+        </div>
+        <div style={{ border: '1px solid' }}>
+          <Checkbox
+            label="isVisible (also set trigger to only manual to programmatically control it)"
+            isChecked={isVisible}
+            onChange={checked => setIsVisible(checked)}
+            aria-label="isVisible"
+            id="is_visible"
+          />
+        </div>
+        <div style={{ border: '1px solid' }}>
+          Entry delay (ms){' '}
+          <TextInput
+            value={entryDelayInput}
+            type="number"
+            onChange={val => setEntryDelayInput(Number(val))}
+            aria-label="entry delay"
+          />
+          Exit delay (ms){' '}
+          <TextInput
+            value={exitDelayInput}
+            type="number"
+            onChange={val => setExitDelayInput(Number(val))}
+            aria-label="exit delay"
+          />
+          Animation duration (ms){' '}
+          <TextInput
+            value={animationDuration}
+            type="number"
+            onChange={val => setAnimationDuration(Number(val))}
+            aria-label="animation duration"
+          />
+        </div>
+        <div style={{ border: '1px solid' }}>
+          flip behavior examples (enableFlip has to be true). "flip" will try to flip the tooltip to the opposite of the
+          starting position. The second option ensures that there are 3 escape positions for every possible starting
+          position (default). This setting is ignored if position prop is set to 'auto'.
+          <Select
+            onToggle={() => setFlipSelectOpen(!flipSelectOpen)}
+            onSelect={(_event, selection) => {
+              setFlipBehavior(selection);
+              setFlipSelectOpen(false);
+            }}
```

Comment:
Update this `onSelect` to be
```
onSelect={(_event: React.MouseEvent | React.ChangeEvent, selection: string | SelectOptionObject) => {
  setFlipBehavior(selection.toString());
  setFlipSelectOpen(false);
}}
```

Reply:
Thank you so much! How did you come to these conclusions that they would solve the issue? I know it's just a `toString()` method but what could I have done better to figure out what the problem was

Reply:
Honesty - it was reading the error messages a bunch of times and testing out many theories 😂 no magic really. brute force.

Reply:
In this case, the type of your setFlipBehavior was never going to match the declared type of the selection param in the onSelect method - so I cast the value passed from the selection (since I really know it will always be a string) so that it matches the type of the state setter method.